### PR TITLE
Fix gio file constructors names

### PR DESCRIPTION
**Pull request by fengalin**
WARNING: don't merge as is. Need to update the gir commit.

Fixes https://github.com/gtk-rs/gtk-rs/issues/502

Depends on https://github.com/gtk-rs/gir/pull/1126.

### CODE REVIEW

File: gio/src/auto/file.rs [hunk 1]
```diff
@@ -56,7 +57,8 @@ impl File {
     }
 
     #[doc(alias = "g_file_new_for_commandline_arg_and_cwd")]
-    pub fn new_for_commandline_arg_and_cwd<P: AsRef<std::ffi::OsStr>, Q: AsRef<std::path::Path>>(
```
Comment (sdroege) [inline gio/src/auto/file.rs:59]: Should maybe autodetect `new_for` like we do `new_from`?

**Comment by sdroege** [comment]
The results look good to me though.

File: gio/src/auto/file.rs (hunk 1 above)
Comment (fengalin) [reply gio/src/auto/file.rs:59]: We already do https://github.com/gtk-rs/gir/blob/b5ff6f8dd8713c3a8d4864f2f975374f0612976b/src/analysis/functions.rs#L558

But that applies only to constructors.

Comment (sdroege) [reply gio/src/auto/file.rs:59]: Oh I see!

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by bilelmoussaoui** [comment]
There are still few others in DBusProxy/DBusConnection/UnixMountEntry that are prefixed with `new_for`, I wonder if it makes sense to handle them here?

**Comment by fengalin** [comment]
`grep -rn new_for **/src/ | wc -l` returns `82` on `gtk-rs`...

**Comment by bilelmoussaoui** [comment]
> `grep -rn new_for **/src/ | wc -l` returns `82` on `gtk-rs`...

most of those are from -sys no?

**Comment by fengalin** [comment]
> most of those are from -sys no?

No, I grepped on `**/src`. I mean you are right, but I thought I was through this. I'll see what I can do later.

**Comment by sdroege** [comment]
> I'll see what I can do later.

Let me know if you want this to be merged first, or do it all at once :)

**Comment by fengalin** [comment]
Some are a bit tricky:

| pkg | struct | funct | comment | proposal |
| ---- | ------- | ------- | ----------- | -------------- |
| `gdk-pixbuf` | `PixbufAnimation` | `new_from_stream_async` | Takes a closure which receives a `Result`, no ret.  | `try_for_stream_async`? |
| '' | '' | `new_from_stream_async_future` | Ret. a `Future` which resolves to a `Result`  | `try_from_stream_async_future`? |
| `gio` | `DbusConnection` | `new_for_address` | Takes a closure which receives a `Result`, no ret.  | `try_for_address`? |
| '' | '' | `new_for_address_future` | Ret. a `Future` which resolves to a `Result`  | `try_for_address_future`? |
| '' | `DBusProxy` | `new_for_bus` | Takes a closure which receives a `Result`, no ret.  | `try_for_bus`? |
| '' | '' | `new_for_bus_future` | Ret. a `Future` which resolves to a `Result`  | `try_for_bus_future`? |
| '' | `Icon` | `new_for_string` | Ret. a `Result`  | `try_for_string`? |
| '' | `UnixMountEntry` | `new_for` | Arg. is a `Path`, Ret. a tuple `(Option<UnixMountEntry>, u64)`  | `try_for`? |
| `pangocairo` | `FontMap` | `new_for_font_type` | Ret. an `Option`  | `try_for_font_type`? |

Note: `gio::DBusConnection::for_address_sync`, which is marked "constructor" and so automatically renamed, also returns a `Result`. We might want `gir` to automatically pick the `try` prefix, not sure if it will be straight forward though...

**Comment by sdroege** [comment]
We don't use the `try_` prefix in autogenerated code for constructors either, so if we start doing that here this also needs some gir changes.

> `UnixMountEntry`

`::try_for_path()` maybe?

**Comment by fengalin** [comment]
> We don't use the try_ prefix in autogenerated code for constructors either, so if we start doing that here this also needs some gir changes.

Ok, I'm not willing to go through this now, so I'll just keep `for_` and `from_`.

> ::try_for_path() maybe?

Ok.

**Comment by sdroege** [comment]
> Ok, I'm not willing to go through this now, so I'll just keep `for_` and `from_`.

Yeah it's all rabbit holes :) Can you create an issue about that so we don't forget?

**Comment by fengalin** [comment]
> Yeah it's all rabbit holes :)

Yeah, even without going through this, I still need another `gir` change :)

> Can you create an issue about that so we don't forget?

https://github.com/gtk-rs/gir/issues/1129

About `UnixMountEntry`, I didn't realized there was also:

```rust
fn new_at<P: AsRef<std::path::Path>>(mount_path: P) -> (Option<UnixMountEntry>, u64)
```

So we might want to keep this one as is for consistency:

```rust
fn new_for<P: AsRef<std::path::Path>>(file_path: P) -> (Option<UnixMountEntry>, u64)
```

What do you think?

**Comment by sdroege** [comment]
> About `UnixMountEntry`, I didn't realize there was also:

I think the C API is badly named :) That should've been `new_for_mount_path()` and `new_for_file_path()` in C because otherwise it's just confusing.

**Comment by fengalin** [comment]
Ok, that was the last issue I raise about these. I'll deliberately ignore any other inconsistencies for now on :)

**Comment by sdroege** [comment]
So let's get this in as-is and worry about the others some other time?

**Comment by fengalin** [comment]
Let me push what I changed and then we'll be all good for now :)

**Review by sdroege — APPROVED** [review]

**Comment by fengalin** [comment]
* Depends on https://github.com/gtk-rs/gir/pull/1130
* I'll update the "gir update" commit before we merge.

**Comment by fengalin** [comment]
Ready to merge! :)
Thanks.

**Pull request merged**